pkg/distributor/distributor.go

```diff
@@ -552,8 +552,14 @@ func (d *Distributor) truncateLines(vContext validationContext, stream *logproto
 	}
 
 	var truncatedSamples, truncatedBytes int
+	maxLineSize := vContext.maxLineSize
+
+	if maxLineSize == 0 {
+		return
+	}
+
 	for i, e := range stream.Entries {
-		if maxSize := vContext.maxLineSize; maxSize != 0 && len(e.Line) > maxSize {
+		if maxSize := maxLineSize; len(e.Line) > maxSize {
```

Comment:
Now you don't need `maxSize` variable at all I think. We can just use `maxLineSize` everywhere. More readable IMHO.

So my understanding is the optimization is just the early return if `maxLineSize==0` correct? Everything else looks same otherwise.

Reply:
the optimization is actually not allocating a new `maxSize` for every entry:
```diff
for i, e := range stream.Entries {
-   if maxSize := vContext.maxLineSize; ...
```